src/hooks-worker-client.coffee

```diff
@@ -1,295 +1,290 @@
+          splittedData = handlerBuffer.split(HANDLER_MESSAGE_DELIMITER)
 
-  # send transaction to the handler
-  message =
-    event: 'afterEach'
-    uuid: uuid
-    data: transaction
+          # add last chunk to the buffer
+          handlerBuffer = splittedData.pop()
 
-  handlerClient.write JSON.stringify message
-  handlerClient.write HANDLER_MESSAGE_DELIMITER
+          messages = []
+          for message in splittedData
+            messages.push JSON.parse message
 
-  # register event for the sent transaction
-  messageHandler = (receivedMessage) ->
-    clearTimeout timeout
-    # workaround for assigning transaction
-    # this does not work:
-    # transaction = receivedMessage.data
-    for key, value of receivedMessage.data
-      transaction[key] = value
-    callback()
+          for message in messages
+            if message.uuid?
+              @emitter.emit message.uuid, message
+            else
+              logger.log 'UUID not present in message: ', JSON.stringify(message, null ,2)
 
-  handleTimeout = () ->
-    transaction.fail = 'Hook timed out.'
-    emitter.removeListener uuid, messageHandler
-    callback()
+    timeout = setTimeout waitForConnect, CONNECT_RETRY
 
-  # set timeout for the hook
-  timeout = setTimeout handleTimeout, HOOK_TIMEOUT
+  registerHooks: (callback) ->
+    eachHookNames = [
+      'beforeEach'
+      'beforeEachValidation'
+      'afterEach'
+      'beforeAll'
+      'afterAll'
+    ]
 
-  emitter.on uuid, messageHandler
+    for eventName in eachHookNames then do (eventName) =>
+      @runner.hooks[eventName] (data, hookCallback) =>
+        # avoiding dependency on external module here.
+        uuid = Date.now().toString() + '-' + Math. random().toString(36).substring(7)
 
-hooks.beforeAll (transactions, callback) ->
-  # avoiding dependency on external module here.
-  uuid = Date.now().toString() + '-' + Math. random().toString(36).substring(7)
+        # send transaction to the handler
+        message =
+          event: eventName
+          uuid: uuid
+          data: data
 
-  # send transaction to the handler
-  message =
-    event: 'beforeAll'
-    uuid: uuid
-    data: transactions
+        @handlerClient.write JSON.stringify message
+        @handlerClient.write HANDLER_MESSAGE_DELIMITER
 
-  handlerClient.write JSON.stringify message
-  handlerClient.write HANDLER_MESSAGE_DELIMITER
+        # register event for the sent transaction
+        messageHandler = (receivedMessage) =>
+          clearTimeout timeout
 
-  # register event for the sent transaction
-  messageHandler = (receivedMessage) ->
-    clearTimeout timeout
-    # workaround for assigning transaction
-    # this does not work:
-    # transaction = receivedMessage.data
-    for value, index in receivedMessage.data
-      transactions[index] = value
-    callback()
+          # workaround for assigning transaction
+          # this does not work:
+          # transaction = receivedMessage.data
```

Comment:
`transaction = receivedMessage.data` is in the comment, but `data` is the name of the variable used below